src/expenses/services/expense.service.js

```diff
@@ -0,0 +1,95 @@
+'use strict';
+
+const { getNewId } = require('../../getNewId');
+
+let expenses = [];
+
+function clearExpenses() {
+  expenses = [];
+}
+
+function getExpenses(userId, from, to, categories) {
+  let filteredExpenses = [...expenses];
+
+  if (userId) {
+    filteredExpenses = filteredExpenses.filter(
+      expense => expense.userId === +userId
+    );
+  }
+
+  if (categories) {
+    filteredExpenses = filteredExpenses.filter(
+      expense => expense.category === categories
+    );
+  }
+
+  if (from) {
+    filteredExpenses = filteredExpenses.filter(
+      expense => expense.spentAt > from
+    );
+  }
+
+  if (to) {
+    filteredExpenses = filteredExpenses.filter(
+      expense => expense.spentAt < to
+    );
+  }
```

Comment:
please use condition in one `.filter` method, because now it will invoke filter a lot of times of there is a multiple filters in query